### ISSUE TITLE
timeout enforcement bug

### ISSUE THREAD

**Issue by boris-dimitrov** [chanzuckerberg/idseq-pipeline#169]
"Command 227 postprocessing after 135.0 seconds."

This shouldn't happen.

Happens here:   https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logEventViewer:group=/aws/batch/job;stream=aegea_batch/default/59b2ad0e-f24d-4b42-8f3c-68bc868e4d14;filter=Command%20227

cc @jshoe

**Comment by boris-dimitrov** [comment]
Looks like a race condition between timer thread cancel and timer thread relaunch.   Need a lock.

```
class Updater(object):

    def __init__(self, update_period, update_function):
        self.update_period = update_period
        self.update_function = update_function
        self.timer_thread = None
        self.t_start = time.time()

    def relaunch(self, initial_launch=False):
        if self.timer_thread and not initial_launch:
            t_elapsed = time.time() - self.t_start
            self.update_function(t_elapsed)
        self.timer_thread = threading.Timer(self.update_period, self.relaunch)
        self.timer_thread.start()

    def __enter__(self):
        self.relaunch(initial_launch=True)
        return self

    def __exit__(self, *args):
        self.timer_thread.cancel()
```

**Comment by boris-dimitrov** [comment]
It's probably mostly harmless aside from leaking threads and polluting the log with print statements.

**Comment by boris-dimitrov** [comment]
Fixed in #174
